indico/modules/categories/forms.py

```diff
@@ -119,57 +142,103 @@ def __init__(self, *args, category, **kwargs):
             self.google_wallet_mode.titles = InheritableConfigMode.get_form_field_titles(parent_configured)
         else:
             self.google_wallet_mode.skip = {InheritableConfigMode.inheriting}
+        if not config.ENABLE_APPLE_PASS:
+            for field in list(self):
+                if field.name.startswith('apple_pass_'):
+                    delattr(self, field.name)
+        elif category.parent:
+            parent_configured = category.parent.effective_apple_pass_config is not None
+            self.apple_pass_mode.titles = InheritableConfigMode.get_form_field_titles(parent_configured)
+        else:
+            self.apple_pass_mode.skip = {InheritableConfigMode.inheriting}
 
     def validate(self, extra_validators=None):
```

Comment:
for the sake of readability I'd prefer adding `_validate_apple_pass` and `_validate_google_wallet` methods and calling those from here. That way you can also avoid re-indenting all the existing code, making it much more easy to review

Reply:
Ok, done.

Reply:
I see that you added `_validate_apple_pass()` and `_validate_google_wallet()` methods, being called from `__init__()`. These methods, however, perform no validation and the logic for validation is still nested in the `validate()` method.

Reply:
Following @ThiefMaster's advice, the methods should be called `_validate_google_wallet_fields` and `_validate_apple_wallet_fields`.

Reply:
@OmeGak sorry, I do not understand. From @ThiefMaster comment above:
`for the sake of readability I'd prefer adding _validate_apple_pass and _validate_google_wallet methods and calling those from here.`

... isn't that what I did?

Reply:
ok... so it belongs to a later comment about using _set_google_wallet_fields .

Reply:
> ... isn't that what I did?

From what I understand, @ThiefMaster commented on the `validate()` method, requested to split its logic into dedicated methods for Google and Apple wallets and then call them from `validate()`. 

From what I see, the `validate()` method executes the Google Wallet logic and the Apple-related methods are per-field. Perhaps this is already ok? I leave it to @ThiefMaster to confirm or provide further guidance.

Reply:
Yes, exactly, I suggested splitting the validation into two separate methods

Reply:
Sorry, but I still do not understand:
The `validate()` method (line 160 of indico/modules/categories/forms.py) contains code ONLY for the validation of the Google Wallet, there is nothing there about the Apple Wallet validation (because they have 2 completely different way of validate data, the Apple validation is done per field), so what should I *split* ?
Eventually I may create a _validate_google_wallet() and move the code from validate() , but that's all.
Is that you want?

Reply:
I got confused because you changed it from

```
if not enabled:
   return

all
the
checks
```

to

```
if enabled:
   all
   the
   checks
```

I think that change can be reverted (it's just adding more changes to the diff which aren't actually needed). No *need* to split it then, even though it might nonetheless increase readability.

Reply:
Done.